Wizard: encode satellite cmd properly

lzap opened the pull request:
I know nothing about the modern JS/TS, review with care. Problems:

* The file /usr/local/sbin/register-satellite is not executable.
* The file does not have a shebang.
* The file contents is for some reason all URL encoded.
* Finally, the file is missing an end newline.

/epic https://issues.redhat.com/browse/HMS-6296

lzap commented on src/Components/CreateImageWizard/utilities/requestMapper.ts:
```diff
@@ -587,7 +587,8 @@ const getCustomizations = (state: RootState, orgID: string): Customizations => {
     files.push({
       path: '/usr/local/sbin/register-satellite',
-      data: btoa(encodeURIComponent(satCmd)),
+      data: btoa("#!/bin/bash\n\n" + satCmd + "\n"),
```
I guess I need to remove it from `getSatelliteCommand` too but I do not know the reason why the value is URL decoded in the first place. @avitova ?

lzap commented:
@avitova had a good point that prefixing/suffixing the string will stack each edit, so removed that. Newline actually does not matter, looks like GNU bash has no problems with that. Also dropped the URI decoding which was not necessary.

avitova commented on src/Components/CreateImageWizard/utilities/requestMapper.ts:
```diff
@@ -587,7 +587,8 @@ const getCustomizations = (state: RootState, orgID: string): Customizations => {
     files.push({
       path: '/usr/local/sbin/register-satellite',
-      data: btoa(encodeURIComponent(satCmd)),
+      data: btoa(satCmd),
```
Are we not adding shebang then?

lzap replied:
No, it is not necessary, I have changed the service unit it now calls `bash -x /the/script`

lzap commented:
Tested new/edit and it works as expected: 

<img width="938" alt="image" src="https://github.com/user-attachments/assets/e43fdf2e-5714-4226-8d80-73a11db924df" />

lzap commented on src/constants.ts:
```diff
@@ -250,7 +250,7 @@ After=osbuild-first-boot.service
 [Service]
 Type=oneshot
-ExecStart=/usr/local/sbin/custom-first-boot
+ExecStart=/bin/bash -x /usr/local/sbin/custom-first-boot
```
This makes the shebang unnecessary. The newline is also not needed, it looks like bash does not care if a script does not have it and EOF will trigger the script as well.

avitova commented:
/retest

avitova commented:
/retest

avitova commented:
/retest ugh

avitova reviewed (approved):
This looks good, let's wait for this ^ and then merge.

lzap commented:
Rebased, moved the explicit `bash` exec to the correct place, thanks that was sneaky!

I am keeping the executable flag check tho, I think it is worth keeping the script executable to be able to re-run it later (e.g. when it fails).

lzap commented:
After chat with @ondrejbudai I was experimenting a bit with something unrelated but I am done, no more experiements let's not play around when in crunch.

```
  [Install]
  WantedBy=basic.target
  WantedBy=multi-user.target
  WantedBy=graphical.target
```

regexowl commented:
The changes look good to me :+1: I'll leave the final word to @avitova as I'm not super familiar with Satellite

avitova reviewed (approved):
Yeah, let's get this in finally. Thanks everyone!:)
